pkgs/misc/uboot/default.nix

```diff
@@ -21,15 +21,17 @@
 , armTrustedFirmwareAllwinnerH616
 , armTrustedFirmwareRK3328
 , armTrustedFirmwareRK3399
+, armTrustedFirmwareRK3588
 , armTrustedFirmwareS905
+, rkbin
 , buildPackages
 }:
 
 let
-  defaultVersion = "2023.07.02";
+  defaultVersion = "2024.01";
   defaultSrc = fetchurl {
     url = "https://ftp.denx.de/pub/u-boot/u-boot-${defaultVersion}.tar.bz2";
-    hash = "sha256-a2pIWBwUq7D5W9h8GvTXQJIkBte4AQAqn5Ryf93gIdU=";
+    hash = "sha256-uZYR8e0je/NUG9yENLaMlqbgWWcGH5kkQ8swqr6+9bM=";
   };
```

Comment:
Can you extract the upgrade to 2024.01 in a different PR? This will need testing and validation on the existing builds too.

Reply:
Understandable, I'll do that tomorrow :)

Reply:
#292757 is ready for review :)

Reply:
Let's mark this PR as draft then until that PR landed, so it doesn't get merged by accident.